invisible_cities/reco/icaro_components_test.py

```diff
@@ -0,0 +1,86 @@
+import numpy         as np
+import numpy.testing as npt
+import pandas        as pd
+
+from hypothesis.strategies import integers
+from hypothesis.strategies import floats
+from hypothesis            import given
+
+
+from .. core.testing_utils import assert_dataframes_equal
+from .. core               import core_functions   as core
+
+from .                     import icaro_components as icarcomp
+
+@given(integers(min_value = 1,
+                max_value = 1e4))
+def test_selection_nS_mask_and_checking_right_output(nsignals):
+    nevt = int(1e4)
+    data = np.concatenate([np.zeros(nevt- nsignals),
+                          np.ones  (nsignals)])
+    np.random.shuffle(data)
+
+    data = pd.DataFrame({'nS1': data, 'event': range(nevt)})
+
+    mask = icarcomp.selection_nS_mask_and_checking(data, icarcomp.type_of_signal.nS1)
+
+    assert np.sum(mask) == nsignals
+
+
+@given(integers(min_value = 1,
+                max_value = 1e4))
+def test_selection_nS_mask_and_checking_self_concatenating(nsignals):
+    nevt = int(1e4)
+    data = np.concatenate([np.zeros(nevt- nsignals),
+                          np.ones  (nsignals)])
+    np.random.shuffle(data)
+
+    data = pd.DataFrame({'nS1': data, 'event': range(nevt)})
+
+    mask    = icarcomp.selection_nS_mask_and_checking(data, icarcomp.type_of_signal.nS1)
+    mask_re = icarcomp.selection_nS_mask_and_checking(data, icarcomp.type_of_signal.nS1, input_mask=mask)
+    npt.assert_equal(mask, mask_re)
+
+
+@given(integers(min_value = 1,
+                max_value = 1e4),
+       integers(min_value = 1,
+                max_value = 1e4))
+def test_selection_nS_mask_and_checking_concatenating(ns1, ns2):
+    nevt = int(1e4)
+    dataS1 = np.concatenate([np.zeros(nevt- ns1),
+                             np.ones (ns1)])
+    dataS2 = np.concatenate([np.zeros(nevt- ns2),
+                             np.ones (ns2)])
+    np.random.shuffle(dataS1)
+    np.random.shuffle(dataS2)
+
+    data   = pd.DataFrame({'nS1': dataS1, 'nS2': dataS2, 'event': range(nevt)})
+    maskS1 = icarcomp.selection_nS_mask_and_checking(data, icarcomp.type_of_signal.nS1)
+    maskS2 = icarcomp.selection_nS_mask_and_checking(data, icarcomp.type_of_signal.nS2, maskS1)
+
+    assert_dataframes_equal(data[maskS2], data[maskS1][data[maskS1].nS2 ==1])
```

Comment:
This feels like it's repeating the source code. Maybe there is something simpler to check like the count on maskS2 <= maskS1 and that every False in maskS1 is still False in maskS2